### DISCUSSION TITLE
Interested in more "pythonic" wrappers?

### DISCUSSION THREAD

**Discussion by ChrisBarker-NOAA** [cubao/concave_hull#19] [Ideas]
I just discovered this recently -- nice work.

As it happens, I did a similar thing a while back, but using cython, rather than pybind11. Only because I'm more familiar with Cython. 

However, I also wrote a little OO python wrapper around it, so that it's easy to simply get the Hull in one step (actually two, but ...).

I submitted a draft of my code to the cpp project, but there has been no response. I jsut pinged it again, but it looks like a pretty dead project :-(

Would you be interested in collaborating on a more pythonic wrapper? the last thing we need is even more forks ...

**Comment by district10** [comment]
@ChrisBarker-NOAA Two solutions to "get the Hull in one step":

1) add scipy dependency and add a function to __init__.py (sounds exactly like your little OO python wrapper)
2) integrate a c++ version of 2d convex hull

I'll try 1) first.

**Comment by district10** [comment]
@ChrisBarker-NOAA How about this: https://github.com/cubao/concave_hull/pull/12

Any suggestions & updates are welcome. I'll merge the PR and release to pypi after your reply.

**Comment by ChrisBarker-NOAA** [comment]
> integrate a c++ version of 2d convex hull

That would be great -- scipy is really common, but it's also a huge dependency. 

It looks like QHULL is set up as a pretty stand alone library -- might not too hard to include it. In fact, a full QHULL wrapper outside of scipy would be pretty nice :-)

Meanwhile I'll take a look at your code to call scipy -- that's a pretty good easy-to-Implement option.

**Comment by district10** [comment]
@ChrisBarker-NOAA I released 0.0.3.

QHULL is much more than "2d convex hull" and pretty heavy itself, so I'd prefer to integrate a small 2d convex hull source just like `concaveman.h`.

**Reply by district10** [reply]
I integrated Grahams-Scan and scipy is not needed anymore.

see 0.0.6 release note: https://concave-hull.readthedocs.io/en/latest/about/release-notes/

**Comment by district10** [comment]
Good news here :)

Configured github workflow to release pypi, and integrated readthedocs.

https://concave-hull.readthedocs.io/en/latest/

**Comment by district10** [comment]
Added support for WGS84 data: https://concave-hull.readthedocs.io/en/latest/wgs84/

**Comment by district10** [comment]
Added interactive demo: https://concave-hull.readthedocs.io/en/latest/wgs84/

Close for now, follow up this issue at Discussion.
